app/controllers/course_wikis_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class CourseWikisController < ApplicationController
+  def update
+    course_id = params[:course_id]
+    wikis = course_wiki_params[:wikis]
+    CourseWiki.where(course_id: course_id).delete_all
+    # clear existing course wikis to update with new ones
+    wikis.each do |wiki|
+      wiki_id = Wiki.get_or_create(language: wiki[:language], project: wiki[:project]).id
+      CourseWiki.create(course_id: course_id, wiki_id: wiki_id)
+    end
+    render json: {}, status: :ok
+  end
+
+  def wikis
```

Comment:
I guess this is intended to be fetched by the frontend, similar to wht we do with `tags` and `campaigns` right now?

The alternative would be to fold this — as well as the update action — into the course update. The question is whether we want changing the wikis to be independent, or wrapped into the data flow for changing other course details (like we currently do with home_wiki). Since it will be a multi-select and not the atomic add/remove actions like with `tags` and `campaigns`, I think the preferable user experience, at least, will be waiting until the `Save` button is clicked to persist the change in wikis... although that could be done with two separate requests, I guess.

On the one hand, course update is already pretty complicated, but on the other hand, it's the one place to go for most changes in course options.

@bwreid what do you think of having a separate controller for CoursesWikis versus rolling wikis into the CoursesController#show and #update actions?

Reply:
Yes, the controller currently is modelled keeping in mind the "Save" is pressed and the wikis are handled. Once we reach a consensus, I will refactor it to suit our needs.

Reply:
Thanks for your patience on this. @AmitJoki My understanding is that we are accessing this information in a limited number of places, correct? Assuming this is true, I do think it would make the most amount of sense to roll this up into the CoursesController. Batching the update is likely a better pattern. 

But, are there cases where we are retrieving this information where we solely want the CourseWiki data? If so, we may want to think differently -- or leave the `show` action here.